chroma_feedback/consumer/razer_chroma/device.py

```diff
@@ -41,7 +49,7 @@ def process_devices(devices : Any, status : str) -> List[Dict[str, Any]]:
 				'consumer': 'razer_chroma',
 				'type': 'device',
 				'name': device.name,
-				'active': pulsate_device(device, color.get_errored()),
+				'active': globals()[effect_method](device, color.get_errored()),
```

Comment:
Please use `pulsate_device(device, color.get_errored()) or breath_device(device, color.get_errored())` ... hopefully one of the methods returns a positive boolean

Reply:
This will require adding try blocks to the _device effect methods, as it will error on the unsupported function, e.g.:

`
Traceback (most recent call last):
  File "./chroma-feedback", line 20, in <module>
    run(program)
  File "/home/jerryq/work/chroma-feedback/chroma_feedback/core.py", line 44, in run
    consumer_result = consumer.process(program, status)
  File "/home/jerryq/work/chroma-feedback/chroma_feedback/consumer/core.py", line 14, in process
    result.extend(consumer.run(status))
  File "/home/jerryq/work/chroma-feedback/chroma_feedback/consumer/razer_chroma/core.py", line 24, in run
    return process_devices(devices, status)
  File "/home/jerryq/work/chroma-feedback/chroma_feedback/consumer/razer_chroma/device.py", line 53, in process_devices
    'active': pulsate_device(device, color.get_failed()) or breath_device(device, color.get_failed()),
  File "/home/jerryq/work/chroma-feedback/chroma_feedback/consumer/razer_chroma/device.py", line 73, in pulsate_device
    return device.fx.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2])
AttributeError: 'RazerFX' object has no attribute 'pulsate'
`

Reply:
What about something like?

```
if device.fx.has('logo_pulsate') and device.fx.has('scroll_pulsate'):
	return device.fx.misc.logo.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2]) and device.fx.misc.scroll_wheel.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2])
if device.fx.has('breath_single')
	return device.fx.breath_single(state['rgb'][0], state['rgb'][1], state['rgb'][2])
return false
```

Reply:
In that case there is no point in separating the functions and I should just call the single method 'pulse_device', handling all possibilities at once, right?  I was thinking that I wanted to let people select the effect they wanted for success or failure, but that's in a separate card.  To do that the driver needs to know what effects are available.

On a separate note, I've found that not all devices have *both* logos and scroll wheels (a non-trivial amount don't have a scroll wheel light).  To that end I'm going to break apart each light (per your recommendation) call with an or, so that mice with only logos can show status as well.

Reply:
Oh, wait, I think I understand what you meant (you have two different function calls in there but that probably wasn't your intention).  The effect code now looks like:

```
	if device.has('brightness'):
		device.brightness = state['brightness'][0]
	if device.fx.has('logo_pulsate') and device.fx.has('scroll_pulsate'):
		return device.fx.misc.logo.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2]) or device.fx.misc.scroll_wheel.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2])
	if device.fx.has('pulsate'):
		return device.fx.pulsate(state['rgb'][0], state['rgb'][1], state['rgb'][2])
	return False
```

Reply:
Yeah, that code looks good. Sorry, the original code mixed pulsate and breath... just finish the PR and I check it out on my devices :-)